File: blog/src/main/java/com/yky/blog/common/entity/Article.java
package com.yky.blog.common.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("article")
public class Article extends BaseEntity {

    @TableId(type = IdType.AUTO)
    private Long id;
    private String title;
    private String cover;
    private String summary;
    private String content;
    private String contentMd;
    private String keywords;
    private String aiDescribe;
    private Long collectionId;
    private Integer status;
    private Integer isStick;
    private Integer isCarousel;
    private Integer carouselSort;
    private Integer isOriginal;
    private String originalUrl;
    private Long viewCount;
    private Long likeCount;
    // IGNORED：updateById 时总是写该列（含 null），使"定时改立即"能清空 publish_time
    @com.baomidou.mybatisplus.annotation.TableField(updateStrategy = com.baomidou.mybatisplus.annotation.FieldStrategy.IGNORED)
    private java.time.LocalDateTime publishTime;
}

File: blog-admin/src/api/article.ts
import request from '@/utils/request'
import type { Tag } from '@/api/tag'

export interface Article {
  id: number
  title: string
  cover: string
  summary: string
  collectionId: number | null
  collectionName: string
  tags: Tag[]
  status: number
  isStick: number
  isCarousel: number
  viewCount: number
  createTime: string
  updateTime: string
}

export interface ArticleDetail extends Article {
  content: string
  contentMd: string
  keywords: string
  aiDescribe: string
  originalUrl: string
  carouselSort: number
  isOriginal: number
  tagIds: number[]
}

export interface ArticleSaveDTO {
  title: string
  cover?: string
  summary?: string
  content?: string
  contentMd: string
  keywords?: string
  aiDescribe?: string
  collectionId?: number | null
  tagIds?: number[]
  status: number
  isStick?: number
  isCarousel?: number
  carouselSort?: number
  isOriginal: number
  originalUrl?: string
}

export interface ArticlePageParams {
  page: number
  size: number
  keyword?: string
  status?: number
}

export const getArticlePage = (params: ArticlePageParams) =>
  request.get('/admin/article/page', { params })

export const getArticleDetail = (id: number) =>
  request.get(`/admin/article/${id}`)

export const saveArticle = (data: ArticleSaveDTO) =>
  request.post('/admin/article', data)

export const updateArticle = (id: number, data: ArticleSaveDTO) =>
  request.put(`/admin/article/${id}`, data)

export const updateArticleStatus = (id: number, status: number) =>
  request.patch(`/admin/article/${id}/status`, { status })

export const deleteArticle = (id: number) =>
  request.delete(`/admin/article/${id}`)
